server/controllers/userController.mjs

```diff
@@ -111,6 +113,62 @@ export const loginUser = async (req, res) => {
   }
 };
 
+export const googleSignIn = async (req, res) => {
+  const { token, userType } = req.body;
+  try {
+    const ticket = await client.verifyIdToken({
+      idToken: token,
+      audience: process.env.GOOGLE_CLIENT_ID,
+    });
+    const payload = ticket.getPayload();
+    const { email, name, picture } = payload;
+
+    let userRecord;
+    try {
+      userRecord = await admin.auth().getUserByEmail(email);
+    } catch (error) {
+      if (error.code === 'auth/user-not-found') {
+        userRecord = await admin.auth().createUser({
+          email,
+          displayName: name,
+          photoURL: picture,
+        });
+
```

Comment:
We don't have "displayName" field in our users collection table.

Reply:
Yes, this is always something confusing, both the display name and photo URL. I will change it.